Python2/3 compatibile (compile and runtime) code

marcin1j opened the pull request:
This makes both ownet and swig (bindings, examples, unit tests) modules Python2/Python3 compatible. Ownet library requires six module (https://pypi.org/project/six/, it's a runtime dependency).

Note that this PR conflicts with python part of #26.

miccoli commented:
THX for the PR: please give me a week or so for the review.

Although it is not strictly python3 compatibility, I would suggest to update  also the ownet `setup.py` to use setuptools and to include the appropriate `install_requires = ['six>= X.Y.Z']`

marcin1j commented:
@miccoli Apparently I've lost track of this PR. Sorry for that.

Personally, after a short period of time I gave up using synchronous ownet module and wrote something with equivalent but asynchronous API, which better suits my needs.

I am still willing to make owfs python3-compatibile. If I'm not mistaken almost two years have passed and it still lacks python3 support. For that reason it's been recently [masked](https://packages.gentoo.org/packages/sys-fs/owfs) and is scheduled to be completely removed from Gentoo's repo!

I can rewrite and test whatever is necessary on top of your changes #26 to make it compile and work correctly on either python3 or python2 and python3.
There's only question if we still needs python2 compatibility at all since it's been unmaintained for nine months now.

miccoli commented:
@marcin1j we usually discourage people from using the "official" owfs python bindings in new projects, since this code is unmaintained and has some serious limitations. The only reason for keeping it alive, is to ease the transition from legacy python2 code to python3, although a rewrite, targeting one of the various python3 libraries available on PyPi, would be in most cases a better choice.

This said I would say that it is better to keep python2 compatibility, which will stay around for a while for running legacy code.

As you can see I pushed a few commits to your repo, addressing some of my concerns. I think that now only some testing is needed to see if all works as expected.

Final touches, like a more clean `configure.ac` and `makefile` structure can be postponed.

marcin1j commented:
I rebased this PR on top of #26. SWIG wrapper now compiles fine on both py2/3.

Note that you need 70027f5 (either cherry-pick 70027f5 or merge this PR with master branch). For testing purposes I rebased my PR on top of c68deb1 but can't do that on github since #26 is still based on 8750e5a.

Since this is the third attempt to upgrade python codebase (#24 addressed `print` syntax is swig module, #26  addressed `raise`/`except` syntax and this one aimed to address both syntax and runtime behavior), it makes sense to combine as much as possible. PR #24 is already merged but #26 is not so feel free to reorganize (cherry-pick or merge) my commits with #26 so that it forms a single branch.

miccoli commented:
@marcin1j tanks for your efforts. I'm a little busy now, but I will for sure merge this and all related PRs.
